contrib/mesos/pkg/scheduler/scheduler.go

```diff
@@ -449,7 +449,7 @@ func (k *KubernetesScheduler) StatusUpdate(driver bindings.SchedulerDriver, task
 func (k *KubernetesScheduler) reconcileTerminalTask(driver bindings.SchedulerDriver, taskStatus *mesos.TaskStatus) {
 	task, state := k.taskRegistry.UpdateStatus(taskStatus)
 
-	if (state == podtask.StateRunning || state == podtask.StatePending) && taskStatus.SlaveId != nil &&
+	if (state == podtask.StateRunning || state == podtask.StatePending) &&
```

Comment:
The `TASK_ERROR` is described here: https://issues.apache.org/jira/browse/MESOS-1143

In short: it's like `TASK_LOST`, but the task should not be restarted by the framework, i.e. "TASK_ERROR for unrecoverable errors".

After reconciliation we got a `TASK_LOST` status with empty executor and empty slaveId.